app/v2/notifications/rest_push.py

```diff
@@ -1,59 +1,106 @@
 
 
 @v2_notification_blueprint.route('/push', methods=['POST'])
 def send_push_notification():
-    return push_notification_helper(push_notification_request, False)
+    return push_notification_helper(push_notification_request)
 
 
 @v2_notification_blueprint.route('/push/broadcast', methods=['POST'])
 def send_push_broadcast_notification():
-    return push_notification_helper(push_notification_broadcast_request, True)
+    return push_notification_helper(push_notification_broadcast_request)
 
 
-def push_notification_helper(schema: dict, is_broadcast: bool):
+def push_notification_helper(schema: dict):
     """
     Note that this helper cannot be called other than as part of a request because it accesses
     the Flask "request" instance.
     """
-
-    if not is_feature_enabled(FeatureFlag.PUSH_NOTIFICATIONS_ENABLED):
-        raise NotImplementedError()
-
     if not authenticated_service.has_permissions(PUSH_TYPE):
         raise BadRequestError(
-            message='Service is not allowed to send {}'.format(get_public_notify_type_text(PUSH_TYPE, plural=True))
+            message='Service is not allowed to send {}'.format(
+                get_public_notify_type_text(PUSH_TYPE, plural=True),
+            ),
+            status_code=403,
         )
 
-    req_json = validate(request.get_json(), schema)
+    validated_payload = validate_push_payload(schema)
 
-    if 'mobile_app' in req_json:
-        app_instance = mobile_app_registry.get_app(MobileAppType[req_json['mobile_app']])
-    else:
-        app_instance = mobile_app_registry.get_app(DEAFULT_MOBILE_APP_TYPE)
+    vetext_formatted_payload = vetext_client.format_for_vetext(validated_payload)
 
     try:
-        vetext_client.send_push_notification(
-            app_instance.sid,
-            req_json['template_id'],
-            req_json['topic_sid'] if is_broadcast else req_json['recipient_identifier']['id_value'],
-            is_broadcast,
-            req_json.get('personalisation'),
+        current_app.logger.debug(
+            'Attempting to call deliver_push celery task with validated payload: %s',
+            vetext_formatted_payload,
         )
-    except VETextBadRequestException as e:
-        raise BadRequestError(message=e.message, status_code=400)
-    except (VETextNonRetryableException, VETextRetryableException):
-        return jsonify(result='error', message='Invalid response from downstream service'), 502
+
+        # Choosing to use the email queue for push to limit the number of empty queues
+        deliver_push.apply_async(
+            args=(vetext_formatted_payload,),
+            queue=QueueNames.SEND_EMAIL,
+        )
+    except (CeleryError, OperationalError):
+        current_app.logger.exception('Failed to enqueue deliver_push request')
+        response = jsonify(result='error', message='VA Notify service impaired, please try again'), 502
```

Comment:
502 indicates a Bad Gateway. I think 503 Service Unavailable is more appropriate if we can't reach celery.

Reply:
Updating